src/resolve/resolved-dns-stub.c

```diff
@@ -462,10 +462,6 @@ static int dns_stub_finish_reply_packet(
                         rcode = DNS_RCODE_SERVFAIL;
         }
 
-        /* Don't set the CD bit unless DO is on, too */
-        if (!edns0_do)
-                cd = false;
-
```

Comment:
why this change?

Reply:
Why transform the packet if we don't have to? I don't think there is any compat concern here, and with this patch we will now update the +cd bit to the "right" value anyway a few lines down, so I don't think this is necessary.

Reply:
where do we do that? dns_stub_finish_reply_packet() is not used in the bypass codepath? or am i blind?

Reply:
Uh you're right. I forgot the context of this change and thought I was looking at a section of patch_bypass in the GH review :|.

I still think the cd bit could be set (if our validation is also disabled), but I will remove this change from the patch.

Reply:
please split this out into a separate commit, it has nothing to do with the proxy